### PR TITLE
feat: Configure publishing for compose multiplatform artifacts

### PR DESCRIPTION
**Pull request by romtsn**
_#skip-changelog_

## :scroll: Description

Configure multiplatform distributions to publish compose artifacts. The maven-publish task spits out everything into the same folder for all targets (common, android and desktop), so we have to manually configure the distribution task to pick up from the correct locations for each target. There's also a different artifact gonna be publish for each target, that's why the release registry has to be updated as well:

https://github.com/getsentry/sentry-release-registry/pull/79

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,6 +1,6 @@
-.PHONY: all clean compile dryRelease update stop checkFormat format api assembleBenchmarkTestRelease assembleUiTestRelease
+.PHONY: all clean compile dryRelease update stop checkFormat format api assembleBenchmarkTestRelease assembleUiTestRelease javadocs
 
-all: stop clean checkFormat compile dryRelease
+all: stop clean javadocs compile
```
Comment (romtsn) [inline Makefile:3]: actually compile (or `./gradlew build`) will run all of the tasks (checkFormat and dryRelease as well), so we were running them twice actuallty. @marandaneto was it on purpose or just and oversight?

**Comment by adinauer** [comment]
cc: @getsentry/open-source can you take a look at the license check please

File: Makefile (hunk 1 above)
Comment (marandaneto) [reply Makefile:3]: IIRC not all tasks were run during `build`, we often found javadocs issues only when releasing, not sure about `checkFormat`.

Comment (romtsn) [reply Makefile:3]: yeah javadoc wasn't running, that's why I added it explicitly here since it's a custom task. But other run just fine

**Review by bruno-garcia — APPROVED** [review]
:shipit:

**Pull request merged**